tests/python/gaia-ui-tests/gaiatest/apps/settings/regions/date_and_time.py

```diff
@@ -21,3 +29,59 @@ def __init__(self, marionette):
     def select_time_format(self, time_format):
         self.marionette.find_element(*self._24h_selector_locator).tap()
         self.select(time_format)
+
+    def set_automatic_time_update(self, state):
+        """
+        toggles 'Set Time Automatically' flag
+        """
+
+        Wait(self.marionette).until(expected.element_displayed(
+            Wait(self.marionette).until(
+                expected.element_present(*self._autotime_enabled_locator))))
+
+        if not state:
+            if self.is_autotime_enabled is True:
+                self.marionette.find_element(*self._autotime_enabled_switch_locator).tap()
+            self.wait_for_condition(lambda m: self.is_autotime_enabled is False)
+        else:
+            if self.is_autotime_enabled is False:
+                self.marionette.find_element(*self._autotime_enabled_switch_locator).tap()
+            self.wait_for_condition(lambda m: self.is_autotime_enabled is True)
+
+    def set_region(self, region):
+        """
+        Setes the time region
+        :param region: Must match the region shown in the list
+        :return:
+        """
+
+        self.marionette.find_element(*self._timezone_region_locator).tap()
+        self.marionette.switch_to_frame()
+        self.wait_for_condition(
+            lambda m: len(self.marionette.find_elements(By.CSS_SELECTOR, '.value-selector-container li')) > 0)
+
+        options = self.marionette.find_elements(By.CSS_SELECTOR, '.value-selector-container li')
+        close_button = self.marionette.find_element(By.CSS_SELECTOR, 'button.value-option-confirm')
+
+        # loop options until we find the match
+        for li in options:
+            if region in li.text:
+                li.tap()
+                break
+        else:
+            raise Exception("Element '%s' could not be found in select wrapper" % region)
+        close_button.tap()
+        self.marionette.switch_to_frame(self.apps.displayed_app.frame)
+
```

Comment:
Isn't this what self.apps.switch_to_displayed_app()) is for?
http://mxr.mozilla.org/gaia/source/tests/python/gaia-ui-tests/gaiatest/gaia_test.py#96